### DISCUSSION TITLE
Using reset() causes all component in <Control> to reset

### DISCUSSION THREAD

**Discussion by mattfranciswork0** [react-hook-form/react-hook-form#6668] [Q&A] [answered]
Hey all,

**Goal** 
When both of the date picker fields are filled, I want to reset the <DatePicker> named "test1" when I click "Test Reset:

**Issue**
When both of the date picker fields are filled, both <DatePicker> reset when I click "Test Reset"

Here is my code sandbox:
https://codesandbox.io/s/react-hook-form-v6-controller-ts-forked-7jjms?file=/src/index.tsx:0-3567

Here is what the bug looks like:
https://gyazo.com/cebceda5e1eaa1bfb4dc3b42394e9435

What seems to be causing the issue?

**Comment by thanh-nguyen-95** [comment, marked as answer]
First of all, `reset()` will replace your whole form values with the object you provided so `reset({ test1: '' })` will make the form values become `{ test1: '' }`, which explains why both input (test1 and test2) seem to be "reset". In your case, you should use `setValue` instead.

Secondly, both Controller AND `register()` are used at the same time, which is not correct. You should only use one of them.

```jsx
<Controller
  control={control}
  // the name of the input will be the name of the Controller's 'name' prop
  name="ReactDatepicker"
  {...register("test1")} // remove this,
  render={(props) => {
    return (
      <ReactDatePicker
        {...props}
        className="input"
        placeholderText="Select date"
        selected={props.value}
      />
    );
  }}
/>
```

Do make sure you use `setValue` with the correct name of input you want to reset, for example:

```js
setValue('ReactDatepicker', '') // this will reset your test1 input
setValue('ReactDatepicker2', '') // this will reset your test2 input
```

By the way, `setValue` doesn't trigger `isDirty` if I remember correctly so if that's what you need then you might want to use reset with the current form values like this:

```js
reset({ ...currentFormValues, ReactDatepicker: '' })
```

I hope this help. Good luck!

**Reply by bluebill1049** [reply]
thank you @thanh-nguyen-95 for helping with all those questions. 🙏
